File: app/src/main/java/abandonedstudio/app/cookingapp/Database/PreparationStepDao.java
package abandonedstudio.app.cookingapp.Database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface PreparationStepDao {

    @Insert
    void insert(PreparationStep preparationStep);

    @Delete
    void delete(PreparationStep preparationStep);

    @Update
    void update(PreparationStep preparationStep);

    @Query("SELECT * FROM preparation_steps WHERE dish_id LIKE :dishId ORDER BY step_number")
    LiveData<List<PreparationStep>> getAllPreparationStepsFromDish(int dishId);
}

File: app/src/main/java/abandonedstudio/app/cookingapp/core/data/local/room/preparation_step/PreparationStepDao.kt
package abandonedstudio.app.cookingapp.core.data.local.room.preparation_step

import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
interface PreparationStepDao {

    @Insert
    suspend fun insert(preparationStep: PreparationStep)

    @Update
    suspend fun update(preparationStep: PreparationStep)

    @Delete
    suspend fun delete(preparationStep: PreparationStep)

    @Query("SELECT * FROM preparation_steps WHERE dish_id LIKE :dishId ORDER BY step_number")
    fun getAllPreparationStepsFromDish(dishId: Int): Flow<List<PreparationStep>>
}